app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,32 @@
+class AnswersController < ApplicationController
+  def new
+    @answer = Answer.new
+  end
+
+  def create
+    answer = Answer.new(answer_params)
+    if answer.save
+      question = Question.find_by(id: answer.question_id)
+      request.xhr? ? render(partial: 'answer', layout: false, locals: {answer: answer}) : redirect_to(question)
+    else
+      flash.now[:error] = answer.errors.full_messages
+      render :new
+    end
+  end
+
+  def edit
+    @answer = Answer.find_by(id: params[:id])
+  end
+
+  def update
+    @answer = Answer.find_by(id: params[:id])
+    @answer.update(answer_params)
```

Comment:
It's always worth checking if this succeeds and telling the user if something went wrong.